### ISSUE TITLE
Transcribe audio execution takes too long and causes failure.

### ISSUE THREAD

**Issue by aef5748** [nextcloud/assistant#151]
### Which version of assistant are you using?

2.1.1

### Which version of Nextcloud are you using?

30.0.1

### Which browser are you using? In case you are using the phone App, specify the Android or iOS version and device please.

Chrome 130.0.6723.60

### Describe the Bug

Upload a voice file of about 25m (voice length is about 1 hour) and send it to localai for conversion. It will show failure after about 5 minutes, but localai is still running, and the results will not be available until about 10 minutes. Can we modify the waiting time?

### Expected Behavior

Wait until the task is completed and return the results

### To Reproduce

1. Upload a large voice file (about 25M or 1 hour of voice content)
2. Use "Transcribe audio" and select this file to send
3. waiting for localai response

**Comment by julien-nc** [comment]
Have you tried changing the "Request timeout" admin setting? Even if it is in the "Text generation" section, it applies to transcription as well.

**Comment by aef5748** [comment]
I tried modifying it, but it didn't work
It still takes about 5 minutes to reply and shows failure.
When I set it to 600s, it will end in about 5 minutes
![Image](https://github.com/user-attachments/assets/0ebe3fc3-80ca-450f-9bd7-3fa8680cf55c)

**Comment by julien-nc** [comment]
The worker needs to be restarted after changing the request timeout value. Are you sure it restarted at least once after you changed the timeout value and before you launched a new transcription task?

**Comment by aef5748** [comment]
Yes, I restarted the worker and nextcloud after the settings were completed.
I have tried the following situations:
1. After restarting the worker, send multiple tasks and repeat them several times.
2. Restart nextcloud and start the worker transfer task
3. Change different options (Is there any difference between these two?)
![Image](https://github.com/user-attachments/assets/1d75485b-673b-4308-8ac9-7c9cd3acc9c5)
5. After modifying the parameter settings, repeat the above actions
![Image](https://github.com/user-attachments/assets/c0dd88ca-1698-4b82-b6cc-65d807858d1f)

No matter how the above is tested, the timeout is usually 5 minutes.
[nextcloud_audio2text_timeout_20241108.log](https://github.com/user-attachments/files/17671593/nextcloud_audio2text_timeout_20241108.log)

**Comment by julien-nc** [comment]
3. "LocalAI's Whisper Speech-To-Text" is the provider for the old transcription API that will be removed in the next release. Both providers are doing the same thing.

The error in your logs shows the timeout of the HTTP client (in integration_openai) is not reached but the peer (server it reaches) ends the connection. So it looks like there is a timeout on the receiving end.

Are you using a proxy to reach LocalAI?
If so, there might be a timeout there.
If not, have you looked for processing time limits in LocalAI?

**Comment by aef5748** [comment]
After removing the proxy settings, it looks normal.
But if I still need proxy settings, how should I set them up?

**Comment by janepie** [comment]
Unfortunately we cannot do configuration support here. You could ask in the [Help forum](https://help.nextcloud.com/) or, if you are a customer, use the official support channels.

I'm closing this now as the app is working :)
